src/Database/Type/SchemaAwareTypeInterface.php

```diff
@@ -0,0 +1,29 @@
+<?php
+declare(strict_types=1);
+
+namespace Cake\Database\Type;
+
+use Cake\Database\DriverInterface;
+use Cake\Database\Schema\TableSchemaInterface;
+
+interface SchemaAwareTypeInterface
```

Comment:
Since we have `TableSchemaAwareInterface` should this be `ColumnSchemaAwareInterface`?

I think we should put the type first not at the end so we don't name interfaces like `SomethingForSomethingInterface`.

Reply:
But is there such a thing as a "column schema"? Aren't columns more like just a part of the "table schema"?

As for the `TypeInterface` trail, I was just following `ExpressionTypeInterface`, but on a second thought this was probably only chosen to avoid confusion with the existing `ExpressionInterface`.

Reply:
I think column types that have a specific expression format have a schema. Consider the GIS types.

Reply:
I couldn't find too much about it, but there's references to column definitions as schemas in for example SQL Server and .NET, so I guess it's kind of a thing.